frontend/src/scenes/web-analytics/WebAnalyticsDataTable.tsx

```diff
@@ -0,0 +1,165 @@
+            return <>Device Type</>
+        default:
+            throw new UnexpectedNeverError(breakdownBy)
+    }
+}
+
+const BreakdownValueCell: QueryContextColumnComponent = (props) => {
+    const { value, query } = props
+    const { source } = query
+    if (source.kind !== NodeKind.WebStatsTableQuery) {
+        return null
+    }
+    if (typeof value !== 'string') {
+        return null
+    }
+
+    return <BreakdownValueCellInner value={value} />
+}
+
+export const webStatsBreakdownToPropertyName = (breakdownBy: WebStatsBreakdown): string => {
+    switch (breakdownBy) {
+        case WebStatsBreakdown.Page:
+            return '$pathname'
+        case WebStatsBreakdown.InitialPage:
+            return '$initial_pathname'
+        case WebStatsBreakdown.InitialReferringDomain:
+            return '$initial_referrer'
+        case WebStatsBreakdown.InitialUTMSource:
+            return '$initial_utm_source'
+        case WebStatsBreakdown.InitialUTMCampaign:
+            return '$initial_utm_campaign'
+        case WebStatsBreakdown.Browser:
+            return '$browser'
+        case WebStatsBreakdown.OS:
+            return '$os'
+        case WebStatsBreakdown.DeviceType:
+            return '$device_type'
+        default:
+            throw new UnexpectedNeverError(breakdownBy)
+    }
+}
+
+const BreakdownValueCellInner = ({ value }: { value: string }): JSX.Element => {
+    return <span>{value}</span>
+}
+
+export const webAnalyticsDataTableQueryContext: QueryContext = {
+    columns: {
+        breakdown_value: {
+            renderTitle: BreakdownValueTitle,
+            render: BreakdownValueCell,
+        },
+        bounce_rate: {
+            title: 'Bounce Rate',
+            render: PercentageCell,
+            align: 'right',
+        },
+        views: {
+            title: 'Views',
+            render: NumericCell,
+            align: 'right',
+        },
+        visitors: {
+            title: 'Visitors',
+            render: NumericCell,
+            align: 'right',
+        },
+    },
+}
+
+export const WebStatsTableTile = ({
+    query,
+    breakdownBy,
+}: {
+    query: DataTableNode
+    breakdownBy: WebStatsBreakdown
+}): JSX.Element => {
+    const { togglePropertyFilter } = useActions(webAnalyticsLogic)
+    const propertyName = webStatsBreakdownToPropertyName(breakdownBy)
+
+    const onClick = useCallback(
+        (breakdownValue: string) => {
+            togglePropertyFilter(propertyName, breakdownValue)
+        },
+        [togglePropertyFilter, propertyName]
+    )
+
+    const context = useMemo((): QueryContext => {
+        const rowProps: QueryContext['rowProps'] = (record: unknown) => {
+            const breakdownValue = getBreakdownValue(record)
+            if (breakdownValue === undefined) {
+                return {}
+            }
+            return {
+                onClick: () => onClick(breakdownValue),
+                className: 'hover:underline cursor-pointer hover:bg-mark',
```

Comment:
Patterns are really important in UX ("don't make me think") and here we're introducing a one we don't have yet (clickable table row), so let's take a moment to consider how to go about this.

For context, we've purposefully stayed away from clickable rows before in favor of the link-name, because there are two downsides: 1. they don't work well for rows which have any other actions, 2. they're not discoverable without hover (there's no expectation of clickability, as most tables don't have clickable rows).

Hence I don't think it's a given that this change will increase discoverability. But we can try it out.

If we try it out, let's make sure that the pattern lives in the table component. Basically I suggest that we don't pass these classes here, but instead if a row has `onClick`, it gets these in `TableRow.tsx`.
As for the specific styling, let's not use `bg-mark`, since it's already used in tables in a different context – it means that the row is statically highlighted. For interactivity `bg-primary-highlight` might be better.
It would also be nice if only the first column was underlined, since all the other ones are just metadata not relevant to the on-click action, but that's more of a nice to have.

Reply:
I think the problem with just having the link-name clickable (like I had before) is that in this case, sometimes the link name is a path. If you have a clickable user id, it's pretty obvious what to expect - clicking it is going to tell me more information about that user. If I have a clickable path, there are 2 potentially obvious things to expect - more info or open that link. Making the row clickable, and whole row highlighted, is to nudge the expectation away from opening the link.

Very happy to move the code for this into the table component - will do that now